pkg/adsc/adsc.go

```diff
@@ -335,23 +336,30 @@ func (a *ADSC) Dial() error {
 }
 
 // Returns a private IP address, or unspecified IP (0.0.0.0) if no IP is available
-func getPrivateIPIfAvailable() net.IP {
+func getPrivateIPIfAvailable() netip.Addr {
 	addrs, _ := net.InterfaceAddrs()
 	for _, addr := range addrs {
-		var ip net.IP
-		switch v := addr.(type) {
-		case *net.IPNet:
-			ip = v.IP
-		case *net.IPAddr:
-			ip = v.IP
-		default:
-			continue
+		var ipAddr netip.Addr
+		addrType := addr.Network()
```

Comment:
why are we switching from the `switch`? It seems a bi more straightforward?

Reply:
Hi @howardjohn, the purpose of the change is to avoid using `net.IP` and use `netip.Addr` instead,

Reply:
Its not avoided... `addr` is still a net.IP, you are just not importing the library explicitly by using string methods (untyped) vs the old approach using typed methods .We lose type safety here.

Reply:
Hi @howardjohn  got it, I get the type safety back, but have to use interface assertion via `net.IPNet` and `net.IPAddr`. There still is no compatible `net.Interfaces` for `net/netip`

Reply:
Any comment?

Reply:
okay, if so, I need to update another place.